### PR TITLE
fix: Fix Feedback for dedicated by using correct Apollo hook

### PR DESCRIPTION
**Pull request by tssweeney**
Internal Jira: https://wandb.atlassian.net/browse/WB-20703

Allows the gorillaGQL url to be passed into browse3.

### CODE REVIEW

File: weave-js/src/apollo.ts [hunk 1]
```diff
@@ -26,7 +26,9 @@ const authMiddleware = new ApolloLink((operation, forward) => {
   });
 });
 
-export const apolloClient = new ApolloClient({
+// This is a hard-coded client that is used in the old weave.wandb.ai and should
+// be removed with that codebase. Using this is INVALID in any proper wandb app.
+export const deprecatedApolloClientDoNotUse = new ApolloClient({
```
Comment (onx2) [inline weave-js/src/apollo.ts:31]: I think if you mark this in JSDocs with `@deprecated` it'll strikethrough in editors like VSCode which could help avoid usage.

File: weave-js/src/components/UserLink.tsx [hunk 2]
```diff
@@ -233,7 +232,7 @@ type UserLinkProps = {
   hasPopover?: boolean; // Can you click to open more a popup
 };
 
-const fetchUser = (userId: string) => {
+const fetchUser = (userId: string, apolloClient: ApolloClient<object>) => {
```
Comment (onx2) [inline weave-js/src/components/UserLink.tsx:235]: Is the concrete type known here so we can avoid `object`?

Comment (tssweeney) [reply weave-js/src/components/UserLink.tsx:235]: nope (: it is just the cache shape which we don't care about - `any` would also suffice

**Review by jamie-rasmussen — APPROVED** [review]

File: weave-js/src/components/UserLink.tsx (hunk 2 above)
Comment (onx2) [reply weave-js/src/components/UserLink.tsx:235]: I think you might be able to omit the type parameter then? 🤔

**Pull request merged**